mob_test.go

```diff
@@ -742,6 +759,95 @@ func TestStartOnUnpushedFeatureBranch(t *testing.T) {
 
 	assertOnBranch(t, "feature1")
 	assertOutputContains(t, output, "Remote branch origin/feature1 is missing")
+	assertOutputContains(t, output, "mob start --create")
+}
+
+func TestStartOnUnpushedFeatureBranchWithUncommitedChanges(t *testing.T) {
+	output, configuration := setup(t)
+	git("checkout", "-b", "feature1")
+	assertOnBranch(t, "feature1")
+	createFile(t, "file.txt", "contentIrrelevant")
+	start(configuration)
+
+	assertOnBranch(t, "feature1")
+	assertOutputContains(t, output, "mob start --include-uncommitted-changes")
+}
+
+func TestStartCreateOnUnpushedFeatureBranch(t *testing.T) {
+	_, configuration := setup(t)
+	configuration.StartCreateRemoteBranch = true
+	git("checkout", "-b", "feature1")
+	assertOnBranch(t, "feature1")
+	start(configuration)
+
+	assertOnBranch(t, "mob/feature1")
+}
+
+func TestStartCreateOnUnpushedFeatureBranchWithUncommitedChanges(t *testing.T) {
+	output, configuration := setup(t)
+	configuration.StartCreateRemoteBranch = true
+	git("checkout", "-b", "feature1")
+	assertOnBranch(t, "feature1")
+	createFile(t, "file.txt", "contentIrrelevant")
+	start(configuration)
+
+	assertOutputContains(t, output, "To start, including uncommitted changes, use")
+	assertOutputContains(t, output, "mob start --create --include-uncommitted-changes")
+}
+
+func TestStartCreateIncludeUncommitedChangesOnUnpushedFeatureBranchWithUncommitedChanges(t *testing.T) {
+	_, configuration := setup(t)
+	configuration.StartCreateRemoteBranch = true
+	configuration.StartIncludeUncommittedChanges = true
+	git("checkout", "-b", "feature1")
+	assertOnBranch(t, "feature1")
+	createFile(t, "file.txt", "contentIrrelevant")
+	start(configuration)
+
+	assertOnBranch(t, "mob/feature1")
+}
+
+func TestStartCreateOnPushedFeatureBranch(t *testing.T) {
+	_, configuration := setup(t)
+	configuration.StartCreateRemoteBranch = true
+	checkoutBranchAndCreateRemoteBranch("feature1")
+	assertOnBranch(t, "feature1")
+	start(configuration)
+
+	assertOnBranch(t, "mob/feature1")
+}
+
+func TestStartCreateOnPushedFeatureBranchWhichIsAhead(t *testing.T) {
+	_, configuration := setup(t)
+	configuration.StartCreateRemoteBranch = true
+	checkoutBranchAndCreateRemoteBranch("feature1")
+	assertOnBranch(t, "feature1")
+	createFile(t, "file.txt", "contentIrrelevant")
+	git("add", ".")
+	git("commit", "-m", "commit ahead")
+	git("push")
+	git("reset", "--hard", "HEAD~1")
+	assertOnBranch(t, "feature1")
+	start(configuration)
```

Comment:
since the tests are getting a bit long, one little thing that might help make those tests easier to read is to move `configuration.StartCreateRemoteBranch = true` down to the act part
```
....

configuration.StartCreateRemoteBranch = true
start(configuration)

assert ...
```
so that you know what this start is about. But I'm not entirely sure, since it really is arranging. Just an Idea.

Reply:
I agree :)